pulp_ansible/tests/functional/api/collection/test_signatures.py

```diff
@@ -1,230 +1,206 @@
+    assert get_content_summary(ansible_repo.to_dict()) == content_summary_dict
+
+    # Extract MANIFEST.json
+    with tarfile.open(collection.filename, mode="r") as tar:
+        filename = tmp_path / "MANIFEST.json"
+        tar.extract("MANIFEST.json", path=tmp_path)
+
+    # Locally sign the collection
+    signing_script_response = sign_with_ascii_armored_detached_signing_service(filename)
+    signature = signing_script_response["signature"]
+
+    # Check that locally produced signature can't be uploaded
+    with pytest.raises(PulpTaskError):
+        task = ansible_collection_signatures_client.create(
+            signed_collection=collection_url, file=signature
+        )
+        monitor_task(task.task)
+
+
+@pytest.mark.pulp_on_localhost
+def test_sign_locally_then_upload_signature(
+    build_and_upload_collection,
+    sign_with_ascii_armored_detached_signing_service,
+    tmp_path,
+    ansible_collections_api_client,
+    ansible_collection_signatures_client,
+):
+    """Test uploading a locally produced Collection Signature."""
+    collection, collection_url = build_and_upload_collection()
+
+    # Extract MANIFEST.json
+    with tarfile.open(collection.filename, mode="r") as tar:
+        filename = tmp_path / "MANIFEST.json"
+        tar.extract("MANIFEST.json", path=tmp_path)
+
+    # Locally sign the collection
+    signing_script_response = sign_with_ascii_armored_detached_signing_service(filename)
+    signature = signing_script_response["signature"]
+
+    # Signature upload
+    task = ansible_collection_signatures_client.create(
+        signed_collection=collection_url, file=signature
+    )
+    sig = monitor_task(task.task).created_resources[0]
+    assert "content/ansible/collection_signatures/" in sig
+
+    # Upload another collection that won't be signed
+    another_collection, another_collection_url = build_and_upload_collection()
+
+    # Check that invalid signatures can't be uploaded
+    with pytest.raises(Exception):
+        task = ansible_collection_signatures_client.create(
+            signed_collection=another_collection_url, file=signature
+        )
+        monitor_task(task.task)
+
+
+@pytest.mark.parallel
+def test_collection_signatures_immutable(ansible_collection_signatures_client):
+    """Test the API doesn't provide an update or partial_update for a Collection Signature objs."""
+    with pytest.raises(AttributeError):
+        ansible_collection_signatures_client.update
+
+    with pytest.raises(AttributeError):
+        ansible_collection_signatures_client.partial_update
+
+
+@pytest.mark.parallel
+def test_collection_signatures_no_deletion(ansible_collection_signatures_client):
+    """Test the API doesn't provide a delete for a Collection Signature obj."""
+    with pytest.raises(AttributeError):
+        ansible_collection_signatures_client.delete
+
+
+# Tests for syncing Collection Signatures
+
+
+def distro_serving_one_signed_one_unsigned_collection(
```

Comment:
Think this needs a fixture mark.

Reply:
It's a fixture not a test, so I'm not sure I can even mark it?

Reply:
Wdym? If it's a pytest fixture doesn't it need the `@pytest.fixture` mark?